licenses/library.go

```diff
@@ -133,9 +128,41 @@ func Libraries(ctx context.Context, classifier Classifier, importPaths ...string
 		}
 		for _, pkg := range pkgs {
 			lib.Packages = append(lib.Packages, pkg.PkgPath)
+			if lib.module == nil {
+				// All the sub packages should belong to the same module.
+				lib.module = newModule(pkg.Module)
+			}
+			if lib.module != nil && lib.module.Path != "" && lib.module.Dir == "" {
+				// A known cause is that the module is vendored, so some information is lost.
+				splits := strings.SplitN(lib.LicensePath, "/vendor/", 2)
```

Comment:
I don't think this is sufficient - there could be non-vendor folder package paths that happen to contain `vendor` in the path. IIUC based on https://go.dev/ref/mod#vendoring we need to check if the the license exists under `${PACKAGE_ROOT}/vendor`

Reply:
TODO: put the comment in the code